upstairs/src/lib.rs

```diff
@@ -3512,26 +3843,24 @@ impl Guest {
 
     pub fn activate(&self, gen: u64) -> Result<(), CrucibleError> {
         let mut waiter = self.send(BlockOp::GoActive { gen });
+        println!("The guest is requesting activation");
         waiter.block_wait()?;
 
-        // XXX Is this the right number of retries? The right delay between
-        // retries?
-        for _ in 0..10 {
+        // XXX This currently blocks forever, but perhaps it should
+        // give up after some time?
+        loop {
```

Comment:
Why did you change this? It should give up after some time :)

Reply:
Yeah.. yeah.. It's like the timeout issue.  How long should the activate wait before it gives up?
What happens if it gives up right before the upstairs returns?  I don't have any answers for these questions.  

Also, the time it takes to activate will be influenced by how long repair takes in the event that the downstairs don't agree.  I can switch it back to 10 seconds, but I'm sure we will be back here 
fussing with this in the future.

Reply:
Switched back in a663fa3

Reply:
> Also, the time it takes to activate will be influenced by how long repair takes in the event that the downstairs don't agree.

This is a great point I didn't consider at all :( Something will have to be done, yeah.